code/modules/projectiles/projectile/beams.dm

```diff
@@ -25,6 +45,13 @@
 	icon_state = "heavylaser"
 	damage = 40
 
+/obj/item/projectile/beam/heavylaser/on_hit(atom/target, blocked = 0)

```

Comment:
Perhaps having firestacks be damage/30 would be nice. Shall I just add a ignitechance variable to give the different weapons different ignite chances?